File: AMW_angular/io/src/app/deployments/logs/deployment-log.ts
export interface DeploymentLog {
  deploymentId: number;
  filename: string;
  content: string;
}

export const toFileName = (deploymentLog: DeploymentLog): string =>
  deploymentLog.filename;

export const filenamePredicate = (filename: string) => (
  deploymentLog: DeploymentLog
): boolean => deploymentLog.filename === filename;

File: AMW_business/src/main/java/ch/puzzle/itc/mobiliar/business/deploy/boundary/DeploymentLog.java
package ch.puzzle.itc.mobiliar.business.deploy.boundary;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class DeploymentLog {

    private long id;
    private String filename;
    private String content;

    public DeploymentLog(Integer id, String filename) {
        this(id, filename, null);
    }
}
